cpp/src/arrow/buffer-builder.h

```diff
@@ -42,7 +49,7 @@ namespace arrow {
 class ARROW_EXPORT BufferBuilder {
  public:
   explicit BufferBuilder(MemoryPool* pool ARROW_MEMORY_POOL_DEFAULT)
-      : pool_(pool), data_(NULLPTR), capacity_(0), size_(0) {}
+      : pool_(pool), data_(&real_location), capacity_(0), size_(0) {}
```

Comment:
I should make this consistent with naming in flatbuffers (either field or constant case, not one in each location)